packages/apollo-language-server/src/project.ts

```diff
@@ -135,20 +151,35 @@ export class GraphQLProject {
 
     this.documentSets = undefined;
     this.engineStats.clear();
+    this.schemaTags = new Map();
     this.documentsByFile = new Map();
     this.setToResolved.clear();
 
-    this.readyPromise = this.loadEngineStats()
-      .then(() => this.scanAllIncludedFiles())
-      .catch(error => {
-        console.error(error);
-      });
+    this.readyPromise = process.env.ENGINE_API_KEY
+      ? this.loadEngineData()
+          .then(() => {
+            this.scanAllIncludedFiles();
+            this._onSchemaTags && this._onSchemaTags(this.schemaTags);
+          })
+          .catch(error => {
+            console.error(error);
```

Comment:
Is this an error that should be send to the loading handler? Otherwise I think the the indicator would get stuck.